src-built-in/components/UIAPI/hooks/linker.ts

```diff
@@ -1,24 +1,120 @@
 
 // Encapsulate the linker initialization and Redux dispatch functions inside the hook
 export const useLinker = () => {
+
     const dispatch = useDispatch();
-    const linker: Linker = useSelector((state: RootState) => state.linker);
+    const state: LinkerState = useSelector((state: RootState) => state.linker);
 
     useEffect(() => {
-		dispatch(actions.init());
+        FSBL.Clients.Logger.system.log("Linker component is mounted. Initializing.");
+        finsembleWindow.addEventListener("blurred", hideWindow);
+        finsembleWindow.addEventListener("shown", focusWindow);
+        setInitialChannels();
+        initAccessibleLinkerMode();
+        FSBL.Clients.RouterClient.addResponder("Finsemble.LinkerWindow.SetActiveChannels", onActiveChannelsChanged);
+
         return () => {
-            dispatch(actions.cleanUp());
+            FSBL.Clients.Logger.system.log("Linker component is unmounted. Cleaning up the event listeners.");
+            finsembleWindow.removeEventListener("blurred", hideWindow);
+            finsembleWindow.removeEventListener("shown", focusWindow);
+            FSBL.Clients.RouterClient.removeEventListener("Finsemble.LinkerWindow.SetActiveChannels", onActiveChannelsChanged);
         }
-	}, []);
+    }, []);
+
+    useEffect(() => {
+        FSBL.Clients.WindowClient.fitToDOM();
+    }, [state])
+
+    const hideWindow = () => {
+        // Must blur or you end up having to click twice to show the window.
+        // If you just hide, the first time the user clicks
+        // on the linker button, the blur will trigger, which will hide the window.
+        finsembleWindow.blur();
+        finsembleWindow.hide();
+    };
+
+    const focusWindow = () => {
+        finsembleWindow.focus();
+    };
+
+    const setInitialChannels = () => {
+        const allChannels: any = FSBL.Clients.LinkerClient.getAllChannels();
+        dispatch(actions.SET_CHANNELS({
+            channels: allChannels,
+        }));
+    };
+
+    /** If user switches the linker channel for different components, this function would be invoked. It will dispatch another
+     *  action to update the linker's state according to the linker setup on the switched component.
+     *
+     * @param err  error from FSBL
+     * @param msg  msg from FSBL
+     */
+    const onActiveChannelsChanged = (err: any, msg: any) => {
+        if (err) FSBL.Clients.Logger.system.error(`Failed to update the linker state.`);
+        dispatch(actions.UPDATE_ACTIVE_CHANNELS({
+            channelNames: msg.data.channels,
+            windowIdentifier: msg.data.windowIdentifier,
+        }));
+        msg.sendQueryResponse(null, null);
+    }
+
+    const initAccessibleLinkerMode = () => {
+        const accessibilityCallback = (err: any, value: boolean) => {
+            if (err) FSBL.Clients.Logger.system.error(`Failed to get accessibleLinker value: ${err}`);
+            dispatch(actions.SET_ACCESSIBILITY({
+                isAccessibleLinker: true,
```

Comment:
Here it should be set to "value" instead of true. Maybe something like ```value ? value : true```

Reply:
Could do !!value